client/my-sites/checkout/composite-checkout/components/jetpack-checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -0,0 +1,205 @@
+	const biennialVariant = variants?.find( ( product ) => product.termIntervalInMonths === 24 );
+
+	if ( ! biennialVariant ) {
+		debug( 'plan in cart has no biennial variant; variants are', variants );
+		return null;
+	}
+
+	if ( ! currentVariant ) {
+		debug( 'plan in cart has no current variant; variants are', variants );
+		return null;
+	}
+
+	if ( biennialVariant.productId === plan.product_id ) {
+		debug( 'plan in cart is already biennial' );
+		return null;
+	}
+
+	const onUpgradeClick = () => {
+		if ( isFormLoading ) {
+			return;
+		}
+		const newPlan = {
+			product_slug: biennialVariant.productSlug,
+			product_id: biennialVariant.productId,
+		};
+		debug( 'switching from', plan.product_slug, 'to', newPlan.product_slug );
+		reduxDispatch(
+			recordTracksEvent( 'calypso_jetpack_checkout_sidebar_upsell_click', {
+				upsell_type: 'biennial-plan',
+				switching_from: plan.product_slug,
+				switching_to: newPlan.product_slug,
+			} )
+		);
+		replaceProductInCart( plan.uuid, newPlan );
+	};
+
+	const percentSavings = getItemVariantDiscountPercentage( biennialVariant, currentVariant );
+	if ( percentSavings === 0 ) {
+		debug( 'percent savings is too low', percentSavings );
+		return null;
+	}
+
+	const isComparisonWithIntroOffer =
+		biennialVariant.introductoryInterval === 2 &&
+		biennialVariant.introductoryTerm === 'year' &&
+		currentVariant.introductoryInterval === 1 &&
+		currentVariant.introductoryTerm === 'year';
+	const currencyConfig = {
+		stripZeros: true,
+		isSmallestUnit: true,
+	};
+
+	const cardTitle = createInterpolateElement(
+		sprintf(
+			// translators: "percentSavings" is the savings percentage for the upgrade as a number, like '20' for '20%'.
+			__( '<strong>Save %(percentSavings)d%%</strong> by paying for two years' ),
+			{ percentSavings }
+		),
+		{ strong: <strong /> }
+	);
+
+	return (
+		<PromoCard title={ cardTitle } className="checkout-sidebar-plan-upsell jetpack">
+			<div className="checkout-sidebar-plan-upsell__plan-grid">
+				<UpsellLine label={ __( 'Yearly plan' ) } boldLabel isTitle />
+
+				<UpsellLine
+					label={ __( 'Year 1' ) }
+					value={ formatCurrency(
+						isComparisonWithIntroOffer
+							? currentVariant.priceInteger
+							: currentVariant.priceBeforeDiscounts,
+						currentVariant.currency,
+						currencyConfig
+					) }
+				/>
+
+				<UpsellLine
+					label={ __( 'Year 2' ) }
+					value={ formatCurrency(
+						currentVariant.priceBeforeDiscounts,
+						currentVariant.currency,
+						currencyConfig
+					) }
+				/>
+
+				<UpsellLine
+					label={ __( 'Total' ) }
+					value={ formatCurrency(
+						currentVariant.priceBeforeDiscounts +
+							( isComparisonWithIntroOffer
+								? currentVariant.priceInteger
+								: currentVariant.priceBeforeDiscounts ),
+						currentVariant.currency,
+						currencyConfig
+					) }
+					boldValue
+				/>
+
+				<UpsellLine label={ __( '2-year plan' ) } boldLabel isTitle />
```

Comment:
IIRC we should spell out numbers and try to avoid  using figures. We've been using "Two-year something" in other places.

Reply:
I agree, I'll get that fixed